### PR TITLE
fix: issue 176 - MINTR missing OHM approval

### PR DESCRIPTION
**Pull request by 0xRusowsky**
Summary: Clearinghouse.sol#claimDefaulted() - Clearinghouse doesn't approve the MINTR to handle tokens in his name, which bricks the entire function.
Issue Link: https://github.com/sherlock-audit/2023-08-cooler-judging/issues/176
Fix Description: Give `MINTR` infinite approval on the OHM token inside the configureDependencies function.

### CODE REVIEW

File: src/Clearinghouse.sol [hunk 1]
```diff
@@ -74,14 +78,16 @@ contract Clearinghouse is Policy, RolesConsumer, CoolerCallback {
...
+        address ohm_,
         address gohm_,
         address staking_,
         address sdai_,
         address coolerFactory_,
         address kernel_
     ) Policy(Kernel(kernel_)) CoolerCallback(coolerFactory_) {
         // Store the relevant contracts.
-        gOHM = ERC20(gohm_);
+        ohm = ERC20(ohm_);
```
Comment (ind-igo) [inline src/Clearinghouse.sol:89]: You can grab the OHM address by calling IStaking(staking_).OHM();

I would move this line below line 91, after initializing staking

File: src/Clearinghouse.sol [hunk 2]
```diff
@@ -103,6 +109,9 @@ contract Clearinghouse is Policy, RolesConsumer, CoolerCallback {
         TRSRY = TRSRYv1(getModuleAddress(toKeycode("TRSRY")));
         MINTR = MINTRv1(getModuleAddress(toKeycode("MINTR")));
         ROLES = ROLESv1(getModuleAddress(toKeycode("ROLES")));
+
+        // Approve MINTR for burning OHM (called here so that it is re-approved on updates)
+        ohm.safeApprove(address(MINTR), type(uint256).max);
```
Comment (ind-igo) [inline src/Clearinghouse.sol:114]: If you are using safeApprove here, you may as well use it everywhere. If not, you can just do regular approval here.

File: src/Clearinghouse.sol [hunk 3]
```diff
@@ -74,14 +78,16 @@ contract Clearinghouse is Policy, RolesConsumer, CoolerCallback {
     // --- INITIALIZATION --------------------------------------------
 
     constructor(
+        address ohm_,
```
Comment (ind-igo) [inline src/Clearinghouse.sol:81]: You don't need to pass in OHM address. You can grab the OHM address by calling staking.OHM();

I would move this line below line 91, after initializing staking

**Review by ind-igo — CHANGES_REQUESTED** [review]

File: src/Clearinghouse.sol [hunk 4]
```diff
@@ -242,9 +248,9 @@ contract Clearinghouse is Policy, RolesConsumer, CoolerCallback {
         });
 
         // Reward keeper.
-        gOHM.transfer(msg.sender, keeperRewards);
+        gohm.transfer(msg.sender, keeperRewards);
         // Unstake and burn the collateral of the defaulted loans.
-        gOHM.approve(address(staking), totalCollateral - keeperRewards);
+        gohm.approve(address(staking), totalCollateral - keeperRewards);
         MINTR.burnOhm(address(this), staking.unstake(address(this), totalCollateral - keeperRewards, false, false));
```
Comment (jkoppel) [inline src/Clearinghouse.sol:254]: I'm expecting to see `ohm.approve(MINTR, ohm.balanceOf(this));`

**Pull request merged**